src/test/java/ch/wisv/areafiftylan/integration/WebTournamentIntegrationTest.java

```diff
@@ -212,4 +234,67 @@ public void testDeleteAllTournamentsAsCommitteeMember() {
             body("message", is("All Tournaments deleted."));
         //@formatter:on
     }
+
+    @Test
+    public void testUpdateTournamentAsUser() {
+        User user = createUser();
+        Tournament tournament = createTournament(),
+                   updateTournament = updateTournament();
+        updateTournament.setId(tournament.getId());
+
+        given()
+            .header(getXAuthTokenHeaderForUser(user))
+            .body(updateTournament)
+            .contentType(ContentType.JSON)
+        .when()
+            .post(TOURNAMENT_ENDPOINT + tournament.getId())
+        .then()
+            .statusCode(HttpStatus.SC_FORBIDDEN);
+
+        assertEquals(1, tournamentRepository.findAll().size());
+        /* toString() is used to force it to check the values, rather than the objects (somehow this failed)
+        * Because tournament is an @data class, the fields are represented correct in the toString() method */
+        assertEquals(tournament.toString(), tournamentRepository.findOne(tournament.getId()).toString());
+    }
+
+    @Test
+    public void testUpdateTournamentAsCommittee() {
+        User committee = createCommitteeMember();
+        Tournament tournament = createTournament(),
+                   updateTournament = updateTournament();
+        updateTournament.setId(tournament.getId());
+
+        given()
+            .header(getXAuthTokenHeaderForUser(committee))
+            .body(updateTournament)
+            .contentType(ContentType.JSON)
+        .when()
+            .post(TOURNAMENT_ENDPOINT + tournament.getId())
+        .then()
+            .statusCode(HttpStatus.SC_CREATED);
+
+        assertEquals(1, tournamentRepository.findAll().size());
+        assertEquals(updateTournament, tournamentRepository.findOne(tournament.getId()));
+    }
+
+    @Test
+    public void testUpdateUnknownTournament() {
+        User admin = createAdmin();
+        Tournament updateTournament = updateTournament();
+        updateTournament.setId(123L);
+
+        assertEquals(0, tournamentRepository.findAll().size());
+
+        given()
+            .header(getXAuthTokenHeaderForUser(admin))
+            .body(updateTournament)
+            .contentType(ContentType.JSON)
+        .when()
+            .post(TOURNAMENT_ENDPOINT + updateTournament.getId())
+        .then()
+            .statusCode(HttpStatus.SC_INTERNAL_SERVER_ERROR)
```

Comment:
This is not a server error, should be handled as a bad request or not found

Reply:
The `TournamentNotFound` exception extends  `AreaFiftyLANException` extends `RunTimeException`. Is there another way to do this?